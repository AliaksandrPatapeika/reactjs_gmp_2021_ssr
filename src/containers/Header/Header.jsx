import React from 'react';
import {useSelector} from 'react-redux';

import Blur from '../../components/Blur';
import HeaderMain from '../HeaderMain';
import HeaderTop from '../HeaderTop';

const Header = () => {
  const activeModalWindow = useSelector((state) => state.movie.activeModalWindow);

  return !activeModalWindow && (
  <header className="headerContainer">
    <Blur>
      <HeaderTop />
      <HeaderMain />
    </Blur>
  </header>
  );
};

export default Header;
