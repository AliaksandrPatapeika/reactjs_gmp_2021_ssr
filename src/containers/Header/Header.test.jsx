import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import Header from './Header';

jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {search: 'movie'}
  })
}));

describe('Header', () => {
  test('renders correctly', () => {
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        activeModalWindow: false
      }
    };
    const store = mockStore(initialState);
    const header = renderer
      .create(
        <Provider store={store}>
          <Header />
        </Provider>
      )
      .toJSON();

    expect(header).toMatchSnapshot();
  });
});
