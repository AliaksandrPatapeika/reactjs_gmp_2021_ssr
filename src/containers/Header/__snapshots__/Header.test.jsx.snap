// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="headerContainer"
>
  <div
    className="blur"
  >
    <div
      className="headerTopContainer"
    >
      <div
        className="logoContainer"
      >
        <span
          className="netflix"
        >
          netflix
        </span>
        <span
          className="roulette"
        >
          roulette
        </span>
      </div>
      <button
        className="addMovieButton"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        + ADD MOVIE
        
      </button>
    </div>
    <div
      className="headerMainContainer"
    >
      <h1
        className="title"
      >
        FIND YOUR MOVIE
      </h1>
      <div
        className="searchContainer"
      >
        <input
          autoComplete="off"
          className="searchInput"
          onChange={[Function]}
          placeholder="What do you want to watch?"
          value=""
        />
        <button
          className="searchButton"
          disabled={false}
          id={null}
          onClick={[Function]}
          type="button"
        >
          SEARCH
          
        </button>
      </div>
    </div>
  </div>
</header>
`;
